src/serializer/ResidualFunctions.js

```diff
@@ -225,7 +225,12 @@ export class ResidualFunctions {
     for (let [funcBody, instances] of functionEntries) {
       let functionInfo = this.residualFunctionInfos.get(funcBody);
       invariant(functionInfo);
-      this._referentialize(functionInfo.unbound, instances);
+      let additionalFunctionInstances = instances.filter((instance) => rewrittenAdditionalFunctions.has(instance.functionValue))
+      if (additionalFunctionInstances && additionalFunctionInstances.length) {
+        invariant(additionalFunctionInstances.length === instances.length, additionalFunctionInstances.length);
```

Comment:
Seems that whether this condition is true or not depends on user input, so should not be an invariant, but rather be logged as an error if violated and important for stuff to work.